microservices/download/utils.js

```diff
@@ -111,175 +107,193 @@ async function displayError(req, res, errMsg) {
 
 async function findLink(os, type) {
   try {
-    let repositoryQuery = `
-      query getRepositoryBuildStatuses {
-        repository(id: 6483909499158528) {
-          builds(branch: "master", last: 10) {
-            edges {
-              node {
-                id
-                status
-              }
-            }
-          }
+
+    let releases = await doRequest();
+
+    // Now these releases should be sorted already, if we find they aren't we might
+    // have to add semver as a dep on this microservice, which is no fun since this
+    // microservice has 0 deps currently. For now lets assume it's a sorted array
+    // This same assumption is made on the `pulsar-updater` core package
+
+    for (let version of releases) {
+      for (let asset of version.assets) {
+
+        let name = asset?.name;
```

Comment:
MINOR CODE NITPICK

Use of optional chaining here isn't really productive with how optional chaining works.

<details>

Optional chaining here checks that trying to access `asset.name` doesn't error of `asset` is undefined, but doesn't ensure that `asset.name` isn't undefined.

So, optional chaining isn't really applicable here, the body of this loop will only execute when iterating over `asset`s that do exist.

</details>

Can remove the optional chaining if you want, it's not hurting anything but it's not doing anything either I think.

```suggestion
        let name = asset.name;
```

And just have the loop handle if `name` turns out to be `undefined`, as you already do just below, that's a good approach already, IMO.

Reply:
You do have a valid point here, without the use of optional chaining, if this item doesn't exist then we just get `undefined` and that is caught during a later check, as we wouldn't ever have an `asset` that itself is undefined.

So you correct this can be removed, although I'll say just for me personally adding things like this helps with not trusting the data we have received to be what it says it is, enforcing the unknown factor of this variable. 

But from a quality standpoint, it's pointless, and could potentially add confusion on how optional chaining operators work, so I'll side with you and remove it

Reply:
I figured out there is one case, if the iterable (usually an array) contains explicit `undefined`, then the optional chaining _would_ be useful.

It happens to be safe not to do the optional chaining since we're getting this from JSON, and `undefined` isn't allowed in JSON.

But your initial instinct was probably about right, my apologies. I think I was a little off-base on this one posting without being expert in it, so if you wish to add it back that is reasonable. (Since we're getting this input data from JSON, it should be okay without, in this case.)